### PR TITLE
Menu, RadioButton: added better theming support.

### PR DESCRIPTION
**Pull request by alansouzati**
#### What does this PR do?
Changes Menu and RadioButton to add better theming support

#### Where should the reviewer start?
Menu.js and StyledRadioButton.js
#### What testing has been done on this PR?
manual
#### How should this be manually tested?
open storybook
#### What are the relevant issues?
https://github.com/grommet/grommet/issues/2336
#### Do the grommet docs need to be updated?
no
#### Should this PR be mentioned in the release notes?
sure
#### Is this change backwards compatible or is it a breaking change?
small breaking change. Had to move `radioButton.border.radius` to `radioButton.check.radius` to align better with CheckBox.

### CODE REVIEW

File: src/js/themes/base.js [hunk 1]
```diff
@@ -517,17 +520,34 @@ export const generate = (baseSpacing = 24, scale = 6) => { // 24
...
       },
+      check: {
+        radius: '100%',
+        // color: { dark: undefined, light: undefined },
+        // extend: undefined,
+      },
+      hover: {
+        border: {
+          color: {
+            dark: css`${props => normalizeColor('white', props.theme)}`,
```
Comment (ericsoderberghp) [inline src/js/themes/base.js]: I think this means we will normalize twice. I think you can just put `dark: 'white'` here.

File: src/js/components/RadioButton/RadioButton.js [hunk 2]
```diff
@@ -34,11 +64,34 @@ class RadioButton extends Component {
...
-          </StyledRadioButtonButton>
+          <StyledRadioButtonBox
+            theme={theme}
+            focus={focus}
+            as={Box}
+            align='center'
+            justify='center'
+            width={theme.radioButton.size}
+            height={theme.radioButton.size}
+            border={{
```
Comment (ericsoderberghp) [inline src/js/components/RadioButton/RadioButton.js]: Can we just set `border={theme.radioButton.border}`?

File: src/js/themes/base.js (hunk 1 above)
Comment (alansouzati) [reply src/js/themes/base.js]: 👍

File: src/js/components/RadioButton/RadioButton.js (hunk 2 above)
Comment (alansouzati) [reply src/js/components/RadioButton/RadioButton.js]: yes! we can actually pass just the color name in the custom theme and rely on normalizeColor to parse that properly

**Pull request merged**